### PR TITLE
breaking: require framework_version, py_version for sklearn

### PR DESCRIPTION
**Pull request by metrizable**
require framework_version and py_version for sklearn

*Issue #, if available:* #1465 

*Description of changes:*

changes include:

* framework_version, py_version required for framework SKLearn
* framework_version, py_version required for framework SKLearnModel
* py_version requires py3 or None, as py2 support dropped
* image_name required if either framework_version or py_version None
* unit and integ testing updates
* doc updates

*Testing done:*

unit:
```
 % tox --parallel 3 tests/unit
✔ OK black-format in 11.287 seconds
✔ OK flake8 in 23.448 seconds
✔ OK twine in 13.898 seconds
✔ OK pylint in 35.064 seconds
✔ OK doc8 in 18.61 seconds
✔ OK py36 in 35.846 seconds
✔ OK sphinx in 1 minute, 26.284 seconds
✔ OK py37 in 36.384 seconds
✔ OK py27 in 1 minute, 50.325 seconds
```

integ:

```
% export IGNORE_COVERAGE=- ; tox -e py37 -- -s tests/integ/test_sklearn_train.py; unset IGNORE_COVERAGE
...
tests/integ/test_sklearn_train.py::test_training_with_additional_hyperparameters
...
Billable seconds: 71
PASSED
tests/integ/test_sklearn_train.py::test_attach_deploy SKIPPED
...
```
## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to any/all clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/sklearn/estimator.py [hunk 1]
```diff
@@ -37,10 +37,10 @@ class SKLearn(Framework):
     def __init__(
         self,
         entry_point,
-        framework_version=defaults.SKLEARN_VERSION,
+        framework_version=None,
```
Comment (ChuyangDeng) [inline src/sagemaker/sklearn/estimator.py:40]: This is defaulting `framework_version` to `None` right? I think we need make it as a required parameter. Or do we have other considerations here?

File: src/sagemaker/sklearn/model.py [hunk 2]
```diff
@@ -62,9 +62,9 @@ def __init__(
         model_data,
         role,
         entry_point,
-        image=None,
+        framework_version=None,
```
Comment (ChuyangDeng) [inline src/sagemaker/sklearn/model.py:65]: same as above.

File: src/sagemaker/sklearn/estimator.py (hunk 1 above)
Comment (metrizable) [reply src/sagemaker/sklearn/estimator.py:40]: good question. the intention is to allow users to specify either `image_name`/`image` or `framework_version` (and `py_version`). we validate either `framework_version` and `py_version` are provided or `image_name` is provided.

File: src/sagemaker/sklearn/model.py (hunk 2 above)
Comment (metrizable) [reply src/sagemaker/sklearn/model.py:65]: see response above

**Pull request merged**